content/posts/Tanmoy_Sarkar_Phase_1_Report.md

```diff
@@ -31,8 +31,11 @@ In the first phase, we have covered
 
 This integration enhances the debugging experience for developers by providing a visual debugging capability directly from their integrated development environment (IDE). It allows anyone to conveniently set breakpoints and debug the application from their IDE. This integration also support docker based environment. By changing configuration in IDE , it can easily connect to TCP based debugger and debug the application easily. [The documentation](https://github.com/tanmoysrt/CircuitVerse/blob/solargraph/yard/documentation/LSP-SETUP.MD) has been updated accordingly.
 
-##### References
-- Demo Video : [Click Here](https://github.com/CircuitVerse/CircuitVerse/assets/57363826/1c0e957e-3eed-4555-8670-27df282786ab)
+**Debugger Preview**
+![ruby-debugger](/images/tanmoy_gsoc23/ruby_debugger.gif)
```

Comment:
Could you make this a video instead of gif, this seems to be 10+MB, automatically loading this gif on the page will slow the page load

Reply:
Should I add link of video only , as in markdown video support is not there ?

Reply:
There are 2 ways you can do this

1. Create your own shortcode named `video.html` in `layouts/shortcodes/video.html`

```html
{{ $videoPath := index .Params 0 }}

<div style="position: relative; padding-bottom: 56.25%; height: 0; overflow: hidden; max-width: 100%; margin: 0 auto;">
  <iframe src="{{ $videoPath }}" width="100%" height="100%" frameborder="0" allowfullscreen style="position: absolute; top: 0; left: 0;"></iframe>
</div>
```
Or use `video` tag instead of `iframe` and get creative with the CSS.

And call it in your markdown as

```bash
{{< video "/images/vaibhav-upreti/ruby_debugger.mp4" >}}
```


2. Use in built hugo `youtube` shortcode that this theme provides by default 

For this youtube video
https://www.youtube.com/watch?v=VmpgOhdMA2g

we can embed it as.... 
```bash
{{< youtube VmpgOhdMA2g >}}
```

Reply:
Thanks @vaibhavupreti

Reply:
@tachyons updated the blog !